### ISSUE TITLE
`invalid-type-guard-call` examples in the docs don't report the error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3735]
```py
from typing import TypeIs

def f(v: object) -> TypeIs[int]: ...

a = [1,2]

f()  # reports `missing-argument` instead
f(*a)  # no error (should report an error because it crashes at runtime, but that's unrelated to `TypeIs`)
f(10)  # no error (false negative)
```

[Playground](https://play.ty.dev/489a618a-6c91-41f5-b62e-05d87c6c731f)

**Comment by DetachHead** [comment]
IMO there's nothing wrong with calling a `TypeGuard`/`TypeIs` function with a literal value, maybe this rule should be deleted (if it hasn't been already?)

**Comment by dhruvmanila** [comment]
I think this seems like a documentation issue, here are the examples in mdtest: https://github.com/astral-sh/ruff/blob/ec35112201c4239b97e188d86423427fd2a6bfa4/crates/ty_python_semantic/resources/mdtest/narrow/type_guards.md#invalid-calls

**Comment by MichaReiser** [comment]
I'll give it a try to see if we can generate mdtests for all our rule documentation so that we catch errors like this automatically

**Comment by DetachHead** [comment]
i'm still a bit confused about this rule. the docs stilll say this:

> Starred (`is_str(*a)`), literal (`is_str(42)`) and other non-symbol-like expressions are invalid as narrowing targets.

but that doesn't seem to actually be the case

i also don't understand the new example:

```py
is_int(value=1)
```

specifying the narrowing target with a keyword arg *does* report the error, regardless of whether the value is a literal expression or not:

```py
is_int(value=value)  # Error
```

i also don't get why this is even an error at all. ty should still be able to tell that this value is a narrowing target. [pyright is fine with it](https://basedpyright.com/?pythonVersion=3.15&typeCheckingMode=all&reportUninitializedInstanceVariable=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgJIDOANFCMQG7ECGANgPrwLEBQXAJscEzlWqGAAo6LAK7EAXFDAAjAFbEAxjACUUALQA%2BQiQoBtUQF1ZXKNZrEYUkCiGpyMRijXEJ04tVGaePgEoYBRvZhl5JVUNTUsbTEEkYVFwmQBeSQjiOOsAYigAURBwEHkURhKwAHdUDDcQNDsoBEZycmJeKEU4KABrYjhq3F4rBJtaBhZ2EjScoA)
